content/hello-world-program.md

```diff
@@ -0,0 +1,328 @@
+
+The focus of this lesson is to build our Solana program, so we’ve gone ahead and provided the client code to invoke our  “Hello, world!” program. The code provided includes a `sayHello` helper function that builds and submits our transaction. We then call `sayHello` in the main function and print a Solana Explorer URL to view our transaction details in the browser.
+
+In the `index.ts` file you should see the following:
+
+```tsx
+import web3 = require("@solana/web3.js");
+import Dotenv from "dotenv";
+Dotenv.config();
+
+let programId = new web3.PublicKey("<YOUR_PROGRAM_ID>");
+
+let payer = initializeKeypair();
+let connection = new web3.Connection(web3.clusterApiUrl("devnet"));
+
+async function main() {
+    await connection.requestAirdrop(payer.publicKey, web3.LAMPORTS_PER_SOL * 1);
+
+    const transactionSignature = await sayHello();
+
+    console.log(
+        `Transaction: https://explorer.solana.com/tx/${transactionSignature}?cluster=custom&customUrl=http%3A%2F%2Flocalhost%3A8899`
+    );
+}
+
+main()
+    .then(() => {
+        console.log("Finished successfully");
+    })
+    .catch((error) => {
+        console.error(error);
+    });
+
+export async function sayHello(): Promise<web3.TransactionSignature> {
+    const transaction = new web3.Transaction();
+
+    const instruction = new web3.TransactionInstruction({
+        keys: [],
+        programId,
+        data: Buffer.alloc(0),
+    });
+
+    transaction.add(instruction);
+
+    const transactionSignature = await web3.sendAndConfirmTransaction(
+        connection,
+        transaction,
+        [payer]
+);
+
+    return transactionSignature;
+}
+
+function initializeKeypair(): web3.Keypair {
+    const secret = JSON.parse(process.env.PRIVATE_KEY ?? "") as number[];
+    const secretKey = Uint8Array.from(secret);
+    const keypairFromSecretKey = web3.Keypair.fromSecretKey(secretKey);
+    return keypairFromSecretKey;
+}
+```
+
+Update the `programId` with the program id you just deployed using Solana Playground.
+
+```tsx
+let programId = new web3.PublicKey("<YOUR_PROGRAM_ID>");
+```
+
+![Gif Solana Playground Program Id](../assets/hello-world-program-id.gif)
+
+Next, install the Node modules with `npm i`. Once those have been installed, we'll need create an `.env` file at the top-level of the client program folder. Add a `PRIVATE_KEY` variable to this file. You can use the Playground wallet you just created.
+
+Now, go ahead and run `npm start`.
+
+![Gif npm start](../assets/hello-world-npm-start.gif)
+
+And finally, copy the transaction URL printed in the console into your browser. Scroll down to see “Hello, world!” under Program Instruction Logs.
+
+![Screenshot Solana Explorer Program Log](../assets/hello-world-program-log.png)
+
+Congratulations, you’ve just successfully built and deployed a Solana program!
+
+# Challenge
+
+Now it’s your turn to build something independently.
```

Comment:
After this sentence add something like: "because we're starting with very simple programs, yours will look almost identical to what we just created. It's useful to try and get to the point where you can write it from scratch without referencing prior code, so try not to copy and paste here."

Reply:
added sentences from above